src/detector/loader.py

```diff
@@ -33,24 +33,25 @@ class Loader:
 
     def __init__(
         self,
-        proto_dirs: Sequence[str],
-        proto_files: Sequence[str] = None,
+        proto_defintion,
+        proto_files=None,
     ):
-        # Check the passing in proto directory is existing or not.
-        for path in proto_dirs:
-            if not os.path.isdir(path):
-                raise TypeError(
-                    f"The directory {path} passed in is not existing. Please check the path."
-                )
-        self.proto_dirs = [dir for dir in proto_dirs]
```

Comment:
The checking is removed, since we already check it when configuring the Options.